### PR TITLE
allows escaped chars to be renderered as html text

### PR DESCRIPTION
**Pull request by nesalang**
fixes issue #1845.

### CODE REVIEW

File: source/shared/cpp/ObjectModel/MarkDownParser.cpp [hunk 1]
```diff
@@ -69,9 +76,15 @@ std::string MarkDownParser::EscapeText()
             break;
         default:
             escaped += m_text.at(i);
+            unescapedCounts++;
             break;
         }
     }
+    
+    if(!m_isEscaped && unescapedCounts != m_text.length())
```
Comment (paulcam206) [inline source/shared/cpp/ObjectModel/MarkDownParser.cpp]: should be a space between if and the opening `(`

File: source/shared/cpp/ObjectModel/MarkDownParser.cpp [hunk 2]
```diff
@@ -35,6 +35,12 @@ bool MarkDownParser::HasHtmlTags()
 {
     return m_hasHTMLTag;
 }
+
+bool MarkDownParser::IsEscaped()
```
Comment (paulcam206) [inline source/shared/cpp/ObjectModel/MarkDownParser.cpp]: this method should be `const`

File: source/shared/cpp/ObjectModel/MarkDownParser.cpp [hunk 3]
```diff
@@ -50,7 +56,8 @@ void MarkDownParser::ParseBlock()
 std::string MarkDownParser::EscapeText()
 {
     std::string escaped;
-
+    unsigned int unescapedCounts = 0;
```
Comment (paulcam206) [inline source/shared/cpp/ObjectModel/MarkDownParser.cpp]: I think something like `nonEscapedCounts` would be a better variable name -- "unescaped" sounds like an action has occurred. :)

File: source/shared/cpp/ObjectModel/MarkDownParser.h [hunk 4]
```diff
@@ -18,12 +18,15 @@ class MarkDownParser
     std::string TransformToHtml();
 
     bool HasHtmlTags();
+
```
Comment (paulcam206) [inline source/shared/cpp/ObjectModel/MarkDownParser.h]: don't need this extra newline

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm [hunk 5]
```diff
@@ -308,7 +308,7 @@ - (void)processTextConcurrently:(std::shared_ptr<BaseCardElement> const &)textEl
             NSDictionary *data = nil;
 
             // use Apple's html rendering only if the string has markdowns
-            if(markDownParser->HasHtmlTags()) {
+            if(markDownParser->HasHtmlTags() || markDownParser->IsEscaped()) {
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm]: do we know if this behavior shows in other platforms?

File: source/shared/cpp/ObjectModel/MarkDownParser.cpp (hunk 3 above)
Comment (nesalang) [reply source/shared/cpp/ObjectModel/MarkDownParser.cpp]: thanks :-)

---
In reply to: [215793995](https://github.com/Microsoft/AdaptiveCards/pull/1865#discussion_r215793995) [](ancestors = 215793995)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm (hunk 5 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm]: the issuer mentioned that he or she tested it also on Desktop and Android, and they don't reproduce, but i'm not certain which version of Destop sample app they are referring to.

---
In reply to: [215794246](https://github.com/Microsoft/AdaptiveCards/pull/1865#discussion_r215794246) [](ancestors = 215794246)

File: source/shared/cpp/ObjectModel/MarkDownParser.h (hunk 4 above)
Comment (nesalang) [reply source/shared/cpp/ObjectModel/MarkDownParser.h]: thanks; it's Xcode doing :-)

---
In reply to: [215794155](https://github.com/Microsoft/AdaptiveCards/pull/1865#discussion_r215794155) [](ancestors = 215794155)

**Pull request merged**